### PR TITLE
AREA-55 - Create view for integration list

### PR DESCRIPTION
**Pull request by Thomaltarix**
I created the integration page.
In this page, you'll be able to see the discord integrations (for now).

Here's a little example of how it looks:
image

> [!Note]
> For now, there isn't discord badges since it's not compulsory and urgent for the MVP

### CODE REVIEW

File: mobile/lib/app/features/integrations/bloc/.gitkeep
Comment (Yann-Masson) [inline mobile/lib/app/features/integrations/bloc/.gitkeep:1]: Maybe you can delete this file if you haven't any bloc state in your feature

File: mobile/lib/app/features/integrations/view/integration_page.view.dart [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+import 'package:triggo/app/widgets/banner.triggo.dart';
+import 'package:triggo/app/widgets/button.triggo.dart';
+import 'package:triggo/mediator/integration.mediator.dart';
+import 'package:triggo/models/integration.model.dart';
+import 'package:triggo/models/integrations/discord.integration.model.dart';
+
+class IntegrationPage extends StatefulWidget {
+  const IntegrationPage({super.key});
+
+  final String title = 'Integration Screen';
```
Comment (Yann-Masson) [inline mobile/lib/app/features/integrations/view/integration_page.view.dart:14]: I think this line is useless.
```suggestion
```

File: mobile/lib/app/features/integrations/view/integration_page.view.dart [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+              children: [
+                TriggoBanner(),
+                const SizedBox(height: 16.0),
+                Align(
+                  alignment: Alignment.centerLeft,
+                  child: Text(
+                    'Integrations',
+                    style: Theme.of(context).textTheme.titleLarge,
+                  ),
+                ),
```
Comment (Yann-Masson) [inline mobile/lib/app/features/integrations/view/integration_page.view.dart:44]: Maybe create a StaelessWidget for this one

File: mobile/lib/app/features/integrations/view/integration_page.view.dart [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+                                    vertical: 8.0, horizontal: 4.0),
+                                child: IntegrationConnectionButton(),
+                              ),
+                            ),
+                          ],
+                        ),
+                        Expanded(
+                            child: IntegrationList(integrations: integrations)),
+                      ],
+                    ),
```
Comment (Yann-Masson) [inline mobile/lib/app/features/integrations/view/integration_page.view.dart:69]: Please create another StaelessWidget for this code.

File: mobile/lib/app/features/integrations/view/integration_page.view.dart [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class IntegrationList extends StatelessWidget {
```
Comment (Yann-Masson) [inline mobile/lib/app/features/integrations/view/integration_page.view.dart:81]: Put an underscore before the classes you use only in this file
```suggestion
class _IntegrationList extends StatelessWidget {
```

File: mobile/lib/app/features/integrations/view/integration_page.view.dart [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+    } else if (!snapshot.hasData || snapshot.data!.isEmpty) {
+      return const NoDataView();
+    } else {
+      return IntegrationListViewContent(integrations: snapshot.data!);
+    }
+  }
+}
+
+class ErrorView extends StatelessWidget {
+  final Object? error;
```
Comment (Yann-Masson) [inline mobile/lib/app/features/integrations/view/integration_page.view.dart:117]: I think you can remove this `?`
```suggestion
  final Object error;
```

File: mobile/lib/models/integrations/discord.integration.model.dart [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    required this.email,
+    required this.displayName,
+    required this.avatarUri,
+    required this.flags,
+  });
+
+  static DiscordIntegration fromDTO(IntegrationDTO dto) {
+    DiscordProperties properties = dto.properties as DiscordProperties;
+    return DiscordIntegration(
+      name: 'Discord',
```
Comment (Yann-Masson) [inline mobile/lib/models/integrations/discord.integration.model.dart:24]: I remember you making a class or enum or listing all the integrations. Can we use this directly? I'm not sure.

**Review by Yann-Masson — CHANGES_REQUESTED** [review]

File: mobile/lib/app/features/integrations/view/integration_page.view.dart (hunk 1 above)
Comment (Thomaltarix) [reply mobile/lib/app/features/integrations/view/integration_page.view.dart:14]: Totally right

File: mobile/lib/app/features/integrations/view/integration_page.view.dart (hunk 5 above)
Comment (Thomaltarix) [reply mobile/lib/app/features/integrations/view/integration_page.view.dart:136]: I can't because the type of `snapshot.error` is Object?

Comment (Yann-Masson) [reply mobile/lib/app/features/integrations/view/integration_page.view.dart:136]: So, if you SURE that snapshot.error is available when you trying to display the error, I think you can give error with `!`. Like this :
```
ErrorView(error: snapshot.error!)
```
I think this work

File: mobile/lib/models/integrations/discord.integration.model.dart (hunk 6 above)
Comment (Thomaltarix) [reply mobile/lib/models/integrations/discord.integration.model.dart:24]: You might think about this method:
```dart
factory IntegrationDTO.fromJson(Map<String, dynamic> json) {
    IntegrationPropertiesDTO properties;
    switch (json['type']) {
      case IntegrationType.discord:
        properties = DiscordProperties.fromJson(json['properties']);
        break;
      default:
        throw Exception('Unknown integration type');
    }
}
```

If not, I'm finding what you're telling me

Comment (Yann-Masson) [reply mobile/lib/models/integrations/discord.integration.model.dart:24]: Mmmhh kinda. But never mind, It's great

File: mobile/lib/app/features/integrations/view/integration_page.view.dart (hunk 5 above)
Comment (Thomaltarix) [reply mobile/lib/app/features/integrations/view/integration_page.view.dart:136]: You're right, didn't think about that !

**Review by Yann-Masson — DISMISSED** [review]

**Review by Yann-Masson — APPROVED** [review]

**Pull request merged**